### ISSUE TITLE
[2.6] AI Civs never construct Submarines

### ISSUE THREAD

**Issue by OmaiGrundles** [LoneGazebo/Community-Patch-DLL#9162]
_1. Mod version, and DirectX version if reporting a graphical issue (X.Y.Z, e.g. 2.6.0, DX 10 & 11):_
Vox Populi with EUI 2.6

_2. Mod list (if using Vox Populi only, leave blank):_
Vox Populi
Wonder Race

_3. Error description:_ 
It is turn 468 on this Prince Difficulty Playthrough, Many Civs have access to every technology, but I noticed that after several wars, I have *never* seen an enemy submarine.

The AI Will make fleets of missile cruisers, and a couple battleships, but never a submarine/nuclear submarine.

I did have some wars during the times where building submarines instead of nuclear ones is possible, and never saw a submarine either, but those wars were short.

The AI has enough aluminum to make jet fighters atleast, usually a large amount of them.
Actually... I don't think I have seen any stealth bombers either (I did see plenty of other bombers though)
Its possible the AI uses up all their aluminum on jet fighters and buildings and never can build anything else?

I am not 100% sure on never seeing a stealth bomber, but I have had huge amounts of naval warfare and never seen an enemy sub.

_4. Steps to reproduce (optional):_
get far enough into the game where the ai can construct these units?

---------------------------
Logs and Save:

[2.6_No_Submarines_StealBombers.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/9453807/2.6_No_Submarines_StealBombers.zip)

**Comment by OmaiGrundles** [comment]
Welp, right after posting this I finally saw an AI stealth bomber, it could be that its because I destroyed 6 or 7 Jet Fighters of theirs the turns before though, and they finally had the aluminum for the bomber?

**Comment by ilteroi** [comment]
you can check their build choices in CityStrategyAIProductionLog.csv (assuming you have logging enabled)

**Comment by OmaiGrundles** [comment]
If i understand the logs correctly (I probably don't), I think it records the 'desire' to make each thing, and ends up making the thing with more points? the points for stuff like nuclear submarines is very low in a lot of cases, one turn it was around 60 points when the highest was 4000 or something.

**Comment by ilteroi** [comment]
yes it can be tricky to interpret ... so you see submarines regularly in the "post" section but their scores are always very low? compared to the top choice or also compared to other naval units?

**Comment by OmaiGrundles** [comment]
![EXCEL_2022-08-31_23-30-58](https://user-images.githubusercontent.com/2267224/187847142-c47826b1-d40e-4040-afec-551b65866e66.png)

Here is the info for one turn on portugul's capital.
On every turn, nuclear submarine gets put into the 'SKIPPED' section (also carrier hits this criteria as well, but I have seen a couple carriers, albeit very few just like the stealth bombers)
It only records a few turns so nuclear sub being on the skipped section every turn could be a coincidence.

**Comment by ilteroi** [comment]
yes those scores certainly look weird ... and the skipping part as well. lategame is not well tested.

**Comment by OmaiGrundles** [comment]
I did a little bit of analysis on the picture above, comparing the 'pre' scores with the 'post' scores:

![EXCEL_2022-09-02_22-44-35](https://user-images.githubusercontent.com/2267224/188257665-9f367ab2-36bf-41f0-a7f4-7f0e41c0ac17.png)

The only unit that gained points out of *all* the choices was the XCOM squad, the original top choice of modern armor got reduced to 24% of its score. 

Also for some reason, mechanical infantry shows up twice.

**Comment by ilteroi** [comment]
re: mech inf - once it's requested for an army, and once as a "free" unit. the interesting question is why the score for the army unit is *lower*

**Comment by ilteroi** [comment]
submarine issue should be fixed, scoring in general is still a mess

**Comment by LoneGazebo** [comment]
> submarine issue should be fixed, scoring in general is still a mess

What specifically is wrong with scoring?

**Comment by ilteroi** [comment]
just take a look at the logs, there are curiosities everywhere ... unit balance is not respected, scores are not really normalized across categories (land unit/naval unit/building/etc).

here's a funny one - AI really loves their missile cruisers

![image](https://user-images.githubusercontent.com/1358966/189522640-0356f3f5-1867-4244-8af0-21551005433d.png)

also they could build nuclear subs now but they never do - because all the uranium is used up ...
